cmd/api/logging/logging.go

```diff
@@ -0,0 +1,162 @@
+func GetKubeArchiveLoggingCredentials() (map[string]string, error) {
+	loggingDir, exists := os.LookupEnv(files.LoggingDirEnvVar)
+	if !exists {
+		return nil, fmt.Errorf("environment variable not set: %s", files.LoggingDirEnvVar)
+	}
+	configFiles, err := files.FilesInDir(loggingDir)
+	if err != nil {
+		return nil, fmt.Errorf("could not read logging credentials: %w", err)
+	}
+	if len(configFiles) == 0 {
+		return nil, fmt.Errorf("logging secret is empty. To configure logging update the kubearchive-logging Secret")
+	}
+
+	loggingConf, err := files.LoggingConfigFromFiles(configFiles)
+	if err != nil {
+		return nil, fmt.Errorf("could not get value for logging credentials: %w", err)
+	}
+	return loggingConf, nil
+}
+
+// LogRetrieval retrieves a middleware function that checks the logging config and when set
+// expects to find a log url in the context from where retrieve and parse logs
+func LogRetrieval(loggingCreds map[string]string, loggingCredsErr error) gin.HandlerFunc {
+	user := loggingCreds[userKey]
+	password := loggingCreds[passwordKey]
+	if user == "" || password == "" || loggingCredsErr != nil {
+		loggingCredsErr = fmt.Errorf(
+			"logging secret values unset. To configure logging update the kubearchive-logging Secret")
+		return func(c *gin.Context) {
+			abort.Abort(c, loggingCredsErr, http.StatusBadRequest)
+		}
+	}
+	// FIXME For now the queries to the logging backend server are done insecurely. Needed for the current test env.
+	// #nosec G402
+	client := http.Client{
+		Transport: otelhttp.NewTransport(&http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true}})}
+
+	return func(c *gin.Context) {
+		logUrl := c.GetString("logURL")
+		jsonPath := c.GetString("jsonPath")
+
+		if logUrl == "" {
+			abort.Abort(c, fmt.Errorf("no log URL found"), http.StatusNotFound)
+			return
+		}
+		slog.Info("Retrieving logs", "logURL", logUrl)
+
+		var jsonPathParser jp.Expr
+		var errJsonPath error
+
+		if jsonPath != "" {
+			jsonPathParser, errJsonPath = jp.ParseString(jsonPath)
+			if errJsonPath != nil {
+				abort.Abort(c, fmt.Errorf("invalid jsonPath %s for url %s", jsonPath, logUrl),
+					http.StatusInternalServerError)
+				return
+			}
+		}
+
+		request, errReq := http.NewRequest(http.MethodGet, logUrl, nil)
+		if errReq != nil {
+			abort.Abort(c, errReq, http.StatusInternalServerError)
+			return
+		}
+
+		request.SetBasicAuth(user, password)
+
+		response, errReq := client.Do(request)
+		if errReq != nil {
+			abort.Abort(c, errReq, http.StatusInternalServerError)
+			return
+		}
+
+		defer response.Body.Close()
+
+		reader := bufio.NewReader(response.Body)
+		line, readErr := readLine(reader)
+		var logsReturned bool
+		for readErr == nil {
+			parsedLine, errParseLine := parseLine(line, jsonPathParser)
+			if errParseLine != nil {
+				abort.Abort(c, errParseLine, http.StatusInternalServerError)
+				return
+			}
+			if parsedLine != "" {
+				logsReturned = true
+				c.JSON(http.StatusOK, parsedLine+"\n")
```

Comment:
Why do we add a jumpline here? I guess its because it failed without it, but why did it fail to create a JSON from a string and needed a jumpline? Or this is because of a representation in your console?

Reply:
It's for readability, the logs come in JSON separated with "\n", so I parse each line and return it again separated with a "\n". Otherwise the readability is pretty bad.

Reply:
I keep using `c.JSON` as we do in the rest of the handlers to keep sending the Content-Type as "application/json" but for the tests that I did with the log-generator (the same from the integration tests) the log output isn't in JSON format, it could be not needed.

I have no strong preference for keeping `c.JSON`, at least for now and the use cases we are thinking of. I can change it to `c.String` if you do.

Reply:
Make it `c.String` we want to avoid serialization/deserialization as much as possible and in this case the impact can be higher as there is no limit to these log lines.

Reply:
Done!